File: clean.py
import re

filepath = 'src/data/mockData.ts'

with open(filepath, 'r', encoding='utf-8') as f:
    content = f.read()

# 1. Single-line replacements: `, en: '...'` or `, en: "..."`
content = re.sub(r',\s*(en|ja|zh)\??:\s*[\'"`].*?[\'"`]', '', content)

# 2. Multi-line replacements: `  en: '...',`
content = re.sub(r'^[ \t]*(en|ja|zh)\??:\s*[\'"`].*?[\'"`],?\r?\n', '', content, flags=re.MULTILINE)

with open(filepath, 'w', encoding='utf-8') as f:
    f.write(content)

print(f"Successfully processed {filepath}")

File: clean.js
const fs = require('fs');
const targetFile = 'src/data/mockData.ts';
let content = fs.readFileSync(targetFile, 'utf-8');

// 1. Single-line objects: replace `, en: '...', ja: '...', zh: '...'`
content = content.replace(/,\s*(en|ja|zh)\??:\s*(['"`]).*?\2/g, '');

// 2. Multi-line properties: remove lines that purely define en/ja/zh
const lines = content.split('\n');
const newLines = lines.filter(line => {
    return !line.match(/^\s*(en|ja|zh)\??:\s*(['"`]).*?\2,?\s*\r?$/);
});

fs.writeFileSync(targetFile, newLines.join('\n'));
console.log('Successfully pruned mockData.ts using Node.js');
